test/xref2/github_issue_941.t/run.t

```diff
@@ -0,0 +1,16 @@
+A quick test to repro the issue found in #941
+
+  $ ocamlc -bin-annot -c foo.mli
+
+  $ odoc compile foo.cmti
+  $ odoc link foo.odoc
+
+  $ odoc html-generate --indent -o html/ foo.odocl
+
+The rendered html
+
+  $ cat html/Foo/index.html | grep "<h3" -A 3
+     <h3 id="splice_me"><a href="#splice_me" class="anchor"></a>Splice me</h3>
+     <p>This is correct <a href="#splice_me">Splice me</a>.</p>
+     <p>This is incorrect <a href="#splice_me">splice_me</a>.</p>
```

Comment:
This is not the expected output. The `<code>` element was fine here, the problem is that the `~text` argument was `None`.
The right value should be generated in `Link.comment_inline_element`.

Reply:
I'v added an ad-hoc case in the pattern matching (up to this point the resolved reference is identical to the "correct case" of the example), let me know if there is a prettier fix